src/components/inputs/input-phone/input-phone.tsx

```diff
@@ -0,0 +1,85 @@
+import { ParseError, parsePhoneNumberWithError } from 'libphonenumber-js';
+import { FC, useState } from 'react';
+
+import CountryPicker from './components/country-picker';
+import { InputPhoneProps } from './input-phone.types';
+import { Box } from '../../../primitives/box/box';
+import { Text } from '../../../primitives/text/text';
+import { InputTextLabel } from '../input-text/components/input-text-label';
+import { InputText } from '../input-text/input-text';
+
+export const InputPhone: FC<InputPhoneProps> = ({
+  onChangePhoneNumber,
+  onCountryPickerPress,
+  countryCode = 'FR',
+  isError,
+  isDisabled,
+  isFocused,
+  description,
+  label,
+  onError,
+  ...props
+}) => {
+  const [inputValue, setInputValue] = useState('');
+
+  const showDescription =
+    (description && !isError) || (isError && inputValue.length > 2);
+
+  const handleOnChangeText = (value: string) => {
+    try {
+      const phoneNumber = parsePhoneNumberWithError(value, countryCode);
+      setInputValue(phoneNumber.formatNational());
+      onChangePhoneNumber?.(phoneNumber);
```

Comment:
Shouldn't it be called also in case of error?

Reply:
We could but the parameter value will be `undefined`. I'm not sure it is relevant